File: framework/extracted/Loenn/entities/realityAnchor.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local realityAnchor = {}

realityAnchor.name = "MaggyHelper/RealityAnchor"
realityAnchor.depth = -100
realityAnchor.fieldInformation = {
    stabilityRadius = {
        minimumValue = 0
    }
}
realityAnchor.placements = {
    {
        name = "normal",
        data = {
            stabilityRadius = 100
        }
    }
}

function realityAnchor.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/reality_anchor/inactive", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {1.0, 1.0, 1.0, 0.8}
    return sprite
end

function realityAnchor.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 40, 32, 40)
end

return realityAnchor

File: framework/extracted/Ahorn/entities/RealityAnchor.jl
module RealityAnchor

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/RealityAnchor" RealityAnchor(x::Integer, y::Integer, stabilityRadius::Integer=100)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(RealityAnchor)
)

function Ahorn.selection(entity::RealityAnchor)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::RealityAnchor, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
